pkgs/by-name/ge/geist-font/package.nix

```diff
@@ -0,0 +1,28 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+}:
+
+stdenv.mkDerivation rec {
+  pname = "geist-font";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "vercel";
+    repo = "geist-font";
+    rev = version;
+    hash = "sha256-WAzNsA80qhBvhutYoHTbp7QbcScNfGWBKIvOA3/ERlc=";
+  };
+
+  postInstall = ''
+    install -Dm444 packages/next/dist/fonts/geist-{mono,sans}/*.woff2 -t $out/share/fonts/woff2
+  '';
```

Comment:
Why is this put in `postInstall` rather than `installPhase`?

Reply:
less boilerplate

Reply:
I don't think boilerplate is a good reason to put commands in semantically incorrect phases.

Reply:
`postInstall` is part of the install phase, there is nothing "semantically incorrect" here

Reply:
I had a look in the documentation. It states no hard rules, but the wording seems to suggest that the pre- and post- hooks are only intended for packages where the default part of the install is already done by the `installPhase`, and downstream users that want to override the package. This package has no such default install action, and the main part of the install happens in `postInstall`.

https://nixos.org/manual/nixpkgs/unstable/#sec-stdenv-phases

> [...] However, typically one only wants to add some commands to a phase, e.g. by defining postInstall or preFixup, as skipping some of the default actions may have unexpected consequences.
>
> When overriding a phase, for example installPhase, it is important to start with runHook preInstall and end it with runHook postInstall, otherwise preInstall and postInstall will not be run. Even if you don’t use them directly, it is good practice to do so anyways for downstream users who would want to add a postInstall by overriding your derivation.

In any case, this makes the package less overridable. But I'll stop messing around here, and rather open a new PR with suggested additions to contributing.

Reply:
It does not make the package "less overrideable".

```nix
pkgs.geist-font.overrideAttrs (oldAttrs: {
  postInstall = oldAttrs.postInstall + ''
    ...
  '';
})
```

And nothing in the docs suggest that `postInstall` is for end users only, nor does the thousands of packages that already use it:

```
~/nixpkgs/ > rg 'postInstall =' | wc -l
3420
```

Reply:
The point of postInstall is to append to the default install phase. Overriding or appending to in this case has the same result here.